test/snippet/core/type_traits/is_instantiable_with.cpp

```diff
@@ -0,0 +1,28 @@
+#include <type_traits>
+
+#include <seqan3/core/type_traits/lazy.hpp>
+#include <seqan3/core/debug_stream.hpp>
+
+template <typename t>
+    requires std::is_integral_v<t>
+struct foo
+{
+    t value;
+};
+
+template <typename t>
+auto bar(t const & v)
+{
+    using cond_t = seqan3::detail::lazy_conditional_t<seqan3::detail::is_instantiable_with_v<foo, t>,
+                                                      seqan3::detail::lazy<foo, t>,
+                                                      t>;
+    return cond_t{v};
+}
+
+int main()
+{
+    auto a = bar(10);
```

Comment:
```suggestion
    foo<int> a = bar(10);
```
This is more expressive about what happens